packages/web/src/components/gcds-sidebar-menu/gcds-sidebar-menu.tsx

```diff
@@ -0,0 +1,155 @@
+  */
+  @State() menuItems = [];
+
+  /**
+  * Current size based on widnow size
+  */
+  @State() menuSize: 'desktop' | 'mobile';
+
+  @Listen("keydown", {target: 'document'})
+  async keyDownListener(e) {
+    if (this.el.contains(document.activeElement)) {
+      handleKeyDownMenu(e, this.el, this.menuItems);
+    }
+  }
+
+  @Listen("gcdsClick", { target: 'document' })
+  async gcdsClickListener(e) {
+    if (this.el.contains(e.target)) {
+      if (e.target.hasAttribute("open")) {
+        await this.updateMenuItemQueue(this.el);
+        (e.target as HTMLGcdsMenuGroupElement).focusTrigger();
+      } else {
+        await this.updateMenuItemQueue(e.target, true);
+        if (e.target.children[0].nodeName == "GCDS-MENU-GROUP") {
+          setTimeout(() => {
+            (e.target.children[0] as HTMLGcdsMenuGroupElement).focusTrigger();
+          }, 10);
+        } else if (e.target.children[0].nodeName == "GCDS-MENU-LINK") {
+          setTimeout(() => {
+            (e.target.children[0] as HTMLGcdsMenuLinkElement).focusLink();
+          }, 10);
+        }
+      }
+    }
+  }
+
+  /*
+  * Observe lang attribute change
+  */
+  updateLang() {
+    const observer = new MutationObserver((mutations) => {
+      if (mutations[0].oldValue != this.el.lang) {
+        this.lang = this.el.lang;
+      }
+    });
+    observer.observe(this.el, observerConfig);
+  }
+
+  /*
+  * Pass new window size: desktop or mobile
+  */
+  @Method()
+  async updateMenuSize(size) {
+    this.menuSize = size;
+  }
+
+  /*
+  * Update item queue for keyboard navigation based on passed element
+  */
+  @Method()
+  async updateMenuItemQueue(el, includeElement?: boolean) {
+    if (includeElement) {
+      let childElements = await getMenuItems(el);
+      this.menuItems = [el, ...childElements];
+    } else {
+      this.menuItems = await getMenuItems(el);
+    }
+  }
+  
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.updateLang();
+
+    const mediaQuery = window.matchMedia('screen and (min-width: 64em)');
+
+    if (mediaQuery.matches) {
+      this.menuSize = 'desktop';
+    } else {
+      this.menuSize = 'mobile';
+      await configureMobileMenu(this.el);
+    }
+  }
+
+  async componentDidLoad() {
+    const mediaQuery = window.matchMedia('screen and (min-width: 64em)');
+    const menu = this.el as HTMLGcdsSidebarMenuElement;
+
+    await this.updateMenuItemQueue(this.el);
+
+    mediaQuery.addEventListener("change", async function(e) {
+      if (e.matches) {
+        menu.updateMenuSize("desktop");
+        await unpackMobileMenu(menu);
+        await menu.updateMenuItemQueue(menu);
+      } else {
+        menu.updateMenuSize("mobile");
+        await configureMobileMenu(menu);
+        await menu.updateMenuItemQueue(menu);
+      }
+    });
+  }
+
+  render() {
+    const { label } = this;
+    return (
+      <Host
+        role="navigation"
+        aria-label={label}
+      >
```

Comment:
Should this all be wrapped in a `nav` tag since it's a navigation menu?

Reply:
I added a `role="navigation"` attribute to the host element to make the `<gcds-sidebar-menu>` element appear as the navigation landmark and not bury the `nav` landmark in the shadowDom.

Do you think it would be better to us a `nav` element in the component?

Reply:
I think for the sake of trying to use semantic elements whenever possible, it would make sense but happy to discuss if you think it's redundant.

Reply:
I will switch the menu components to use `nav` elements